torch/_dynamo/guards.py

```diff
@@ -623,11 +622,16 @@ def source_ref(source):
             source_ref,
             combine_scopes(f_globals, f_locals),
             self,
-            renames=True,
+            local=True,
         )
         global_builder = GuardBuilder(
-            self.id_ref, source_ref, f_globals, self, renames=False
+            self.id_ref, source_ref, f_globals, self, local=False
         )
+        # We need to transplant a copy here, because some guards
+        # might get a cross ref between local and global, like L['mod_name'][G['some_key']]
+        # the inverse is illegal.
+        if "G" in global_builder.scope:
+            local_builder.scope["G"] = global_builder.scope["G"]
         # source_ref can cause a cycle, make sure we break it with weakref
         w_local = weakref.ref(local_builder)
         w_global = weakref.ref(global_builder)
```

Comment:
Scroll down a bit to line 660 (new) and there's this code

```

        assert not (set(local_builder.argnames) & set(global_builder.argnames))
        # see parallel handling of ".0" / "___implicit0" in _eval_frame.c
        largs = [a for a in local_builder.scope.keys() if a == "___implicit0"]
        largs += [a for a in local_builder.argnames if a != "___implicit0"]
        largs += ["**___kwargs_ignored"]
        args = ",".join(largs)
```

We've removed .0 renaming to __implicit0 so something needs to happen to this code too

Reply:
Good eye